### DISCUSSION TITLE
q-select any way to detect click of down-arrow?

### DISCUSSION THREAD

**Discussion by CrawfordW** [quasarframework/quasar#11092] [General - Components / Directives / etc] [answered]
When the option list is long, I want to filter so the drop-down list only shows options that could match what the user has typed so far. But if she then clicks the down-arrow, I need to show the entire option list, not the filtered short list.
I think I could do this if there was a unique event for clicking the down arrow. But maybe there's a better way.
Any suggestions? Or is this a new feature request?

**Comment by metalsadman** [comment]
you can listen for `@keydown="handler"` then check for the `event.key === 'ArrowDown'` in your handler. https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#example

**Reply by CrawfordW** [reply]
Sorry, perhaps I wasn't clear... I meant using the mouse to click the down-arrow icon in the select box, not using a keyboard arrow key.
Currently q-select seems to generate the same event whether the user clicks in the input box, or clicks on the down arrow icon. It would be helpful to have a unique event for the down-arrow.

**Reply by hawkeye64** [reply]
If you figure out the position, you can use `clientX` and `clientY` from the event.

**Comment by metalsadman** [comment, marked as answer]
> Sorry, perhaps I wasn't clear... I meant using the mouse to click the down-arrow icon in the select box, not using a keyboard arrow key. Currently q-select seems to generate the same event whether the user clicks in the input box, or clicks on the down arrow icon. It would be helpful to have a unique event for the down-arrow.

what Jeff suggested, not all browsers has displays an arrow (afaik, there's no native event to catch the actual event you're looking for.), what you could do is check for when the element is scrolling or has scrolled. check out qselect's `@virtual-scroll` https://codepen.io/metalsadman/pen/wvqdmyz?editors=101

**Reply by CrawfordW** [reply]
OK thanks for the suggestions.
I built a component that combines q-select and q-tree so the drop-down can be tree-structured while the q-select menu is only used for auto-type. I could use that for my purpose, but I need a version that lives in a table and automatically expands to fill the available space in the column. Unfortunately I haven't been able to figure out how to make the main div that contains the q-select and q-menu do this.

**Reply by metalsadman** [reply]
i would advice to keep to the topic of the thread and post other query into a different one.
